### PR TITLE
Adding the TERMUX_PKG_PROVIDES value for the packages to work correctly (for pacman format).

### PR DESCRIPTION
**Pull request by Maxython**
This change only affects pacman packages.

### CODE REVIEW

File: packages/dropbear/build.sh [hunk 1]
```diff
@@ -9,6 +9,9 @@ TERMUX_PKG_SHA256=48235d10b37775dbda59341ac0c4b239b82ad6318c31568b985730c788aac5
 TERMUX_PKG_DEPENDS="termux-auth, zlib"
 TERMUX_PKG_CONFLICTS="openssh"
 TERMUX_PKG_BUILD_IN_SRC=true
+if [ $TERMUX_PACKAGE_FORMAT = "pacman" ]; then
+	TERMUX_PKG_PROVIDES="openssh"
```
Comment (xeffyr) [inline packages/dropbear/build.sh:13]: It doesn't provide OpenSSH but has symlinks `scp` and `ssh` which ARE NOT compatible with OpenSSH.

File: packages/nodejs-lts/build.sh [hunk 2]
```diff
@@ -16,6 +16,9 @@ TERMUX_PKG_SUGGESTS="clang, make, pkg-config, python"
 TERMUX_PKG_RM_AFTER_INSTALL="lib/node_modules/npm/html lib/node_modules/npm/make.bat share/systemtap lib/dtrace"
 TERMUX_PKG_BUILD_IN_SRC=true
 TERMUX_PKG_HOSTBUILD=true
+if [ $TERMUX_PACKAGE_FORMAT = "pacman" ]; then
+	TERMUX_PKG_PROVIDES="nodejs"
```
Comment (xeffyr) [inline packages/nodejs-lts/build.sh:20]: This can be acceptable.

File: packages/nodejs/build.sh [hunk 3]
```diff
@@ -16,6 +16,9 @@ TERMUX_PKG_SUGGESTS="clang, make, pkg-config, python"
 TERMUX_PKG_RM_AFTER_INSTALL="lib/node_modules/npm/html lib/node_modules/npm/make.bat share/systemtap lib/dtrace"
 TERMUX_PKG_BUILD_IN_SRC=true
 TERMUX_PKG_HOSTBUILD=true
+if [ $TERMUX_PACKAGE_FORMAT = "pacman" ]; then
+	TERMUX_PKG_PROVIDES="nodejs-lts"
```
Comment (xeffyr) [inline packages/nodejs/build.sh:20]: But this not. You should not describe the latest version of NodeJS as LTS (aka long-term-support).

File: packages/openssh/build.sh [hunk 4]
```diff
@@ -8,6 +8,9 @@ TERMUX_PKG_SRCURL=https://fastly.cdn.openbsd.org/pub/OpenBSD/OpenSSH/portable/op
 TERMUX_PKG_SHA256=4590890ea9bb9ace4f71ae331785a3a5823232435161960ed5fc86588f331fe9
 TERMUX_PKG_DEPENDS="libandroid-support, ldns, openssl, libedit, termux-auth, krb5, zlib"
 TERMUX_PKG_CONFLICTS="dropbear"
+if [ $TERMUX_PACKAGE_FORMAT = "pacman" ]; then
+	TERMUX_PKG_PROVIDES="dropbear"
```
Comment (xeffyr) [inline packages/openssh/build.sh:12]: Same as with Dropbear package case. Dropbear has symlinks `ssh` and `scp` which are made for "convenience" but really they are pointing on utilities that are not compatible with OpenSSH.

You should not specify `TERMUX_PKG_PROVIDES` there.

**Review by xeffyr — COMMENTED** [review]
These changes breaking package relations.

For Dropbear/OpenSSH, see how this was implemented by Arch Linux: https://github.com/archlinux/svntogit-community/blob/packages/dropbear/trunk/PKGBUILD#L53-L54

Notes:
* Specify `TERMUX_PKG_CONFLICTS` when package has conflicting files, e.g. packages provide utilities with same names.
* Specify `TERMUX_PKG_PROVIDES` to create alternate alias for package, e.g. when package has been renamed but you don't want break dependencies or want let users install package by specifying old name.

File: packages/dropbear/build.sh (hunk 1 above)
Comment (Maxython) [reply packages/dropbear/build.sh:13]: which is better to specify `scp` or `ssh`?

**Comment by Maxython** [comment]
Should I create another pull requests?

**Comment by xeffyr** [comment]
Why you need to specify "provides" field at least for `openssh` and `dropbear`? Package manager **should** throw an error when you are attempting to have `openssh` and `dropbear` installed together. These packages are not interchangeable and can't not provide each other.

Dropbear has `ssh` (real utility name: `dbclient`) and `scp` (force-enabled at build time, not available by default in dropbear) links in package on our behalf and that should not be changed.

At max you can specify `TERMUX_PKG_PROVIDES="nodejs"` for package `nodejs-lts` for both pacman and debian package types.

**Comment by Maxython** [comment]
> Why you need to specify "provides" field at least for openssh and dropbear?

This is necessary so that packages with openssh "or" dropbear dependencies can work with dropbear(for example [autossh](https://github.com/termux/termux-packages/blob/master/packages/autossh/build.sh#L10)). As a result, when compiling for the pacman format, they have the following dependency format `TERMUX_PKG_DEPENDS="openssh"`, because of which they will not work with dropbear. For these packages to work with dropbear, dropbear must be set to `TERMUX_PKG_PROVIDES="openssh"` so that pacman understands that dropbear has the same function as openssh. Now we are not talking about virtual packages, we are talking about the correct work of dependencies with "or". 

> for both pacman and debian package types.

This `TERMUX_PKG_PROVIDES` should only work for pacman format.

**Pull request closed without merging**